pkg/lib/saml/samlprotocol/samlprotocolhttp/result.go

```diff
@@ -36,7 +36,8 @@ func (s *SAMLResult) IsInternalError() bool {
 
 type SAMLErrorResult struct {
 	SAMLResult
-	Cause error
+	Cause        error
+	IsUnexpected bool
```

Comment:
I think we should use `errors.As` to check if it is a SAML error. If it is, then it is an expected error, otherwise, it is unexpected. So remove this field.

Reply:
As discussed before, errors from service layer (i.e. SAMLError) does not determine the the error in handler layer.
The same saml error could have different meaning in different handler.
For example, if we got `service_provider_not_found` in login or metadata, it is expected error, but if we got it in login_finish, it is unexpected error (because the service provider id is from session).
So only the handler should translate any error to expected to unexpected error.

Reply:
Then I suggest we replace `NewSAMLErrorResult` with `NewExpectedSAMLErrorResult` and `NewUnExpectedSAMLErrorResult`. The positional argument of `true, false` is very easy to miss.